SalesforceCommerceCloud/pwa-kit
# storefront preview : add promotional pricing to PDP

for storefront preview, adds promotional price to PDP, ported from v3 PR #1455 

# Description

This PR creates a DisplayPrice component to show discount price and strikethrough base price, if applicable.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- added components/display-price/index.jsx
- components/product-tile/index.jsx
- hooks/use-add-to-cart-modal.js
- added hooks/use-product-pricing.js
- partials/product-view/index.jsx
- added theme/components/project/display-price.js
- theme/index.js

# How to Test-Drive This PR

- `npm start` on retail app
- Go to this product: http://localhost:3000/global/en-GB/product/25695436M?color=JJP49XX&size=9SM&pid=701644154707M
- Discount price should show up with basePrice is being strikethrough

![Screenshot 2023-10-06 at 5 28 20 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/375787/7d70af2e-b66d-48a9-9778-183b656c14db)

![Screenshot 2023-10-06 at 5 31 30 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/375787/20121a94-3133-49ca-a79a-07f7c27724b6)

![Screenshot 2023-10-06 at 5 32 39 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/375787/1598c3c7-4bd7-4c96-a82f-9dff87a677b8)


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
